src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentDiagnosticFactory.cs

```diff
@@ -421,5 +421,33 @@ public static RazorDiagnostic Create_RefSuppressFieldRequiredForGeneric(SourceSp
         {
             return RazorDiagnostic.Create(RefSuppressFieldRequiredForGeneric, source ?? SourceSpan.Undefined);
         }
+
+        public static readonly RazorDiagnosticDescriptor ComponentNamesCannotStartWithLowerCase =
+            new RazorDiagnosticDescriptor(
+                $"{DiagnosticPrefix}10013",
+                () => "Component '{0}' starts with a lowercase character. Component names cannot start with a lowercase character.",
+                RazorDiagnosticSeverity.Error);
+
+        public static RazorDiagnostic Create_ComponentNamesCannotStartWithLowerCase(string componentName, SourceSpan? source = null)
+        {
+            return RazorDiagnostic.Create(
+                ComponentNamesCannotStartWithLowerCase,
+                source ?? SourceSpan.Undefined,
+                componentName);
+        }
+
+        public static readonly RazorDiagnosticDescriptor UnexpectedMarkupElement =
+            new RazorDiagnosticDescriptor(
+                $"{DiagnosticPrefix}10014",
+                () => "Found markup element with unexpected name '{0}'. If this is intended to be a component, add a @using directive for its namespace.",
+                RazorDiagnosticSeverity.Warning);
```

Comment:
Since Todd is back now does he know about our ask for warnings? I know we met with WTE but wanted to make sure that one requirement was still being tracked on their end.

Reply:
Yeah. I mentioned it in our meeting and in the email I sent